apps/node-fastify/src/crons/refresh-stale-subscriptions.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Orb currently has no webhook whenever a billing cycle is reset, so subscription billing cycle information may be outdated.
+ *
+ * As a temporary solution, we use a CRON to look at active subscriptions with a billing cycle end date in the past.
+ *
+ * Orb provides an API that takes multiple customer ids when looking for subscriptions.
+ */
+
+import { OrbSync } from 'orb-sync-lib';
+import { logger } from '../utils/logger';
+
+export async function refreshStaleSubscriptions(orbSync: OrbSync) {
+  logger.info(`Starting to refresh stale subscriptions`);
+
+  const customerIdsWithOutdatedSubscriptionsData = await orbSync.postgresClient
+    .selectMany<{ customer_id: string }>(
+      `
+        SELECT customer_id FROM orb.subscriptions
+        WHERE status = 'active'
+        AND current_billing_period_end_date < :now
```

Comment:
We might need to look into query performance, but probably not an urgent issue

Reply:
Yeah, like you said, for now performance shouldn't be a big issue. I created a Notion task to check and if applicable improve the performance.
https://www.notion.so/supabase/Orb-Sync-Engine-Stale-Subscription-CronJob-Query-Performance-1135004b775f808a86fbdb5dc95a9bcc?pvs=4

Reply:
yeah good enough for me